virtool/downloads/api.py

```diff
@@ -204,3 +205,30 @@ async def download_sequence(req):
     return web.Response(text=fasta, headers={
         "Content-Disposition": f"attachment; filename={filename}"
     })
+
+
+@routes.get("/download/subtraction/{subtraction_id}")
+async def download_subtraction(req):
+    """
+    Download a FASTA file for the given subtraction.
+
+    """
+    db = req.app["db"]
+    subtraction_id = req.match_info["subtraction_id"]
+
+    document = await db.subtraction.find_one(subtraction_id)
+
+    if document is None:
+        return virtool.api.response.not_found()
+
+    if "has_file" not in document or document["has_file"] is False:
+        return virtool.api.response.not_found("Subtraction FASTA file not found")
+
+    path = os.path.join(
+        virtool.subtractions.utils.join_subtraction_path(req.app["settings"], subtraction_id),
+        "subtraction.fa.gz"
+    )
+
+    return web.FileResponse(path, headers={
+        "Content-Type": "application/gzip"
```

Comment:
Use `Content-Disposition` to change file name to something more meaningful. Currently, downloaded file is called `<subtraction_id>.gz`.

```
// Template
subtraction-<id>-<name>.fa.gz

// Actual
subtraction-12345-arabidopsis-thaliana.fa.gz 
```

Lowercase subtraction `name` and replace spaces with `-`.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition